language_service/src/qsc_utils.rs

```diff
@@ -59,5 +64,15 @@ pub(crate) fn find_item<'a>(compilation: &'a Compilation, id: &ItemId) -> Option
     } else {
         &compilation.unit.package
     };
-    package.items.get(id.item)
+    package.items.get(id.item).map(|item| {
+        (
+            item,
+            item.parent
+                .and_then(|parent_id| package.items.get(parent_id))
+                .and_then(|parent| match &parent.kind {
+                    qsc::hir::ItemKind::Namespace(namespace, _) => Some(namespace.name.clone()),
+                    _ => None,
+                }),
+        )
+    })
```

Comment:
IMO this logic looks a bit one-off and belongs more in hover.rs (where you call `find_item`). You don't even use the namespace return value at the other callsite in display.rs.

Reply:
Since I'd need access to the `package` to do this logic, and that is most of what this function does, I'd have to have a hover-specific version of this function that is exactly what's in the PR now. Maybe an alternative is for the `find_item` to return the item and the `package` in a tuple?

Reply:
I've tried the approach to return the package with the item. Let me know if this is better.

Reply:
Oh, I see. Yes I like what you came up with - somehow it seems more reusable for the future (though somewhat hypothetical at the moment)